kubernetes/linux/setup.sh

```diff
@@ -9,88 +9,71 @@ else
+    wget "https://github.com/microsoft/Docker-Provider/releases/download/official%2Fmdsd%2F1.17.1%2Frpm/azure-mdsd_1.17.1-build.master.377_aarch64.rpm" -O azure-mdsd.rpm
 fi
+sudo tdnf install -y azure-mdsd.rpm
 
-/usr/bin/dpkg -i $TMPDIR/azure-mdsd*.deb
+# /usr/bin/dpkg -i $TMPDIR/azure-mdsd*.deb
 cp -f $TMPDIR/mdsd.xml /etc/mdsd.d
 cp -f $TMPDIR/envmdsd /etc/mdsd.d
 rm /usr/sbin/telegraf
 
+mdsd_version=$(sudo tdnf list installed | grep mdsd | awk '{print $2}')
+echo "Azure mdsd: $mdsd_version" >> packages_version.txt
+
 # log rotate conf for mdsd and can be extended for other log files as well
 cp -f $TMPDIR/logrotate.conf /etc/logrotate.d/ci-agent
 
 #download inotify tools for watching configmap changes
-sudo apt-get update
-sudo apt-get install inotify-tools -y
+sudo tdnf check-update -y
+sudo tdnf install inotify-tools -y
 
 #used to parse response of kubelet apis
 #ref: https://packages.ubuntu.com/search?keywords=jq
-sudo apt-get install jq=1.5+dfsg-2 -y
+sudo tdnf install jq-1.6-1.cm2 -y
 
 #used to setcaps for ruby process to read /proc/env
-sudo apt-get install libcap2-bin -y
-
-TELEGRAF_VERSION=1.25.2
-wget https://dl.influxdata.com/telegraf/releases/telegraf-${TELEGRAF_VERSION}_linux_${ARCH}.tar.gz
-tar -zxvf telegraf-${TELEGRAF_VERSION}_linux_${ARCH}.tar.gz
+sudo tdnf install libcap -y
 
-mv /opt/telegraf-${TELEGRAF_VERSION}/usr/bin/telegraf /opt/telegraf
-
-chmod 544 /opt/telegraf
+sudo tdnf install telegraf-1.25.2 -y
+telegraf_version=$(sudo tdnf list installed | grep telegraf | awk '{print $2}')
+echo "telegraf $telegraf_version" >> packages_version.txt
+mv /usr/bin/telegraf /opt/telegraf
 
 # Use wildcard version so that it doesnt require to touch this file
 /$TMPDIR/docker-cimprov-*.*.*-*.*.sh --install
+docker_cimprov_version=$(sudo tdnf list installed | grep docker-cimprov | awk '{print $2}')
+echo "DOCKER_CIMPROV_VERSION=$docker_cimprov_version" >> packages_version.txt
 
-#download and install fluent-bit(fluent-bit)
-wget -qO - https://packages.fluentbit.io/fluentbit.key | sudo apt-key add -
-sudo echo "deb https://packages.fluentbit.io/ubuntu/bionic bionic main" >> /etc/apt/sources.list
-sudo apt-get update
-sudo apt-get install fluent-bit=2.0.5 -y
+#install fluent-bit
+sudo tdnf install fluent-bit-2.0.9 -y
+echo "$(fluent-bit --version)" >> packages_version.txt
 
 # fluentd v1 gem
-gem install fluentd -v "1.14.6" --no-document
+# gem install fluentd -v "1.14.6" --no-document
+sudo tdnf install rubygem-fluentd-1.14.6 -y
+echo "$(fluentd --version)" >> packages_version.txt
 fluentd --setup ./fluent
-gem install gyoku iso8601 --no-doc
+
+gem install gyoku iso8601 bigdecimal --no-doc
 gem install tomlrb -v "2.0.1" --no-document
 
 
 rm -f $TMPDIR/docker-cimprov*.sh
-rm -f $TMPDIR/azure-mdsd*.deb
 rm -f $TMPDIR/mdsd.xml
 rm -f $TMPDIR/envmdsd
-rm -f $TMPDIR/telegraf-*.tar.gz
 
 # remove build dependencies
-sudo apt-get remove gcc make -y
-sudo apt autoremove -y
+sudo tdnf remove gcc make -y
```

Comment:
this is quite old and there might be an opportunity to remove other packages as well

Reply:
We will be copying on those packages needed anyway in distroless. So yes no need to remove this separately here.